### PR TITLE
Add password policy resource

### PR DESCRIPTION
**Pull request by emichaeli**
Add password policy resource to the Terraform provider: https://api.sumologic.com/docs/#tag/passwordPolicy

### CODE REVIEW

**Review by prashant-trivedi — COMMENTED** [review]
In general lgtm. Will leave approval to Sean and Vishal.

File: sumologic/resource_sumologic_password_policy.go
Comment (sumovishal) [inline sumologic/resource_sumologic_password_policy.go:13]: We should allow importing existing password policy. Don't forget to add import acceptance test as well.

File: sumologic/resource_sumologic_password_policy.go
Comment (sumovishal) [inline sumologic/resource_sumologic_password_policy.go:18]: Let's define a top level variable (like `DefaultPasswordPolicy`) with default values and use them here and in the test code (`newDefaultPasswordPolicy`).

**Review by sumovishal — COMMENTED** [review]
Left a couple of comments, rest lgtm.

**Review by sumovishal — APPROVED** [review]
Approving to unblock merging. Please address https://github.com/SumoLogic/terraform-provider-sumologic/pull/161/files#r583752869 before merging.

File: sumologic/sumologic_password_policy.go
Comment (seansain) [inline sumologic/sumologic_password_policy.go:24]: rename the function `ResetPasswordPolicy`

**Pull request merged**
